### DISCUSSION TITLE
Design space termination (xtol)

### DISCUSSION THREAD

**Discussion by sam-c-smt** [anyoptimization/pymoo#344] [Q&A]
Hi,

When using DesignSpaceTermination with multi-variable design space, which, if any, of the following is it looking at:
1. Average design space variable change over all variables for **all candidates** in population.
2. Average design space variable change over all variables for **best candidate**.
3. Maximum designs space change of any **one variable** within **any candidate**.

Thanks in advance!

**Comment by blankjul** [comment]
Let us say we have in iteration t the solutions `S_t`. And then an iteration later `S_(t+1)`. Then the termination averages the distances from each solution in `S_(t+1)` to the closest solution in `S_t`.

I hope this helps!
